Common/GL/GLInterface/EGL.cpp

```diff
@@ -0,0 +1,228 @@
+// Copyright 2012 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <array>
+#include <cstdlib>
+
+#include "Common/GL/GLInterface/EGL.h"
+#include "Common/Log.h"
+
+// Show the current FPS
+void cInterfaceEGL::Swap()
+{
+	eglSwapBuffers(egl_dpy, egl_surf);
+}
+void cInterfaceEGL::SwapInterval(int Interval)
+{
+	eglSwapInterval(egl_dpy, Interval);
+}
+
+void* cInterfaceEGL::GetFuncAddress(const std::string& name)
+{
+	return (void*)eglGetProcAddress(name.c_str());
+}
+
+void cInterfaceEGL::DetectMode()
+{
+	if (s_opengl_mode != MODE_DETECT)
+		return;
+
+	EGLint num_configs;
+	bool supportsGL = false, supportsGLES2 = false, supportsGLES3 = false;
+	std::array<int, 3> renderable_types = {
+		EGL_OPENGL_BIT,
+		(1 << 6), /* EGL_OPENGL_ES3_BIT_KHR */
+		EGL_OPENGL_ES2_BIT,
+	};
+
+	for (auto renderable_type : renderable_types)
+	{
+		// attributes for a visual in RGBA format with at least
+		// 8 bits per color
+		int attribs[] = {
+			EGL_RED_SIZE, 8,
+			EGL_GREEN_SIZE, 8,
+			EGL_BLUE_SIZE, 8,
```

Comment:
Do we need/want to care about depth/stencil for non-buffered folks?

-[Unknown]